pdns/slavecommunicator.cc

```diff
@@ -242,12 +242,12 @@ static bool processRecordForZS(const DNSName& domain, bool& firstNSEC3, DNSResou
 static vector<DNSResourceRecord> doAxfr(const ComboAddress& raddr, const DNSName& domain, const TSIGTriplet& tt, const ComboAddress& laddr,  scoped_ptr<AuthLua4>& pdl, ZoneStatus& zs)
 {
   vector<DNSResourceRecord> rrs;
-  AXFRRetriever retriever(raddr, domain, tt, (laddr.sin4.sin_family == 0) ? NULL : &laddr, ((size_t) ::arg().asNum("xfr-max-received-mbytes")) * 1024 * 1024);
+  AXFRRetriever retriever(raddr, domain, tt, (laddr.sin4.sin_family == 0) ? NULL : &laddr, ((size_t) ::arg().asNum("xfr-max-received-mbytes")) * 1024 * 1024, ::arg().asNum("axfr-fetch-timeout"));
   Resolver::res_t recs;
   bool first=true;
   bool firstNSEC3{true};
   bool soa_received {false};
-  while(retriever.getChunk(recs)) {
+  while(retriever.getChunk(recs, 0, ::arg().asNum("axfr-fetch-timeout"))) {
```

Comment:
The `::arg().asNum(..)` call is not free - perhaps you can stick it in a `uint16_t` at the beginning of the function?

Reply:
Also we prefer using `nullptr` over `0` when dealing with pointers